File: SDKs/amazonaws.com/cloudfront/2020-05-31/typescript/src/sdk/models/shared/publickeysummarylist.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// PublicKeySummaryList
/** 
 * Contains information about a public key.
**/
export class PublicKeySummaryList extends SpeakeasyBase {
  @SpeakeasyMetadata()
  comment?: string;

  @SpeakeasyMetadata()
  createdTime: Date;

  @SpeakeasyMetadata()
  encodedKey: string;

  @SpeakeasyMetadata()
  id: string;

  @SpeakeasyMetadata()
  name: string;
}

File: SDKs/amazonaws.com/cloudfront/2020-05-31/python/src/sdk/models/shared/publickeysummarylist.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional


@dataclasses.dataclass
class PublicKeySummaryList:
    r"""PublicKeySummaryList
    Contains information about a public key.
    """
    
    created_time: datetime = dataclasses.field()
    encoded_key: str = dataclasses.field()
    id: str = dataclasses.field()
    name: str = dataclasses.field()
    comment: Optional[str] = dataclasses.field(default=None)
    
